From 2a97a6c2763714655387188a2333c7fc8538052b Mon Sep 17 00:00:00 2001
From: Xylar Asay-Davis <xylarstorm@gmail.com>
Date: Fri, 27 Jan 2023 14:17:16 -0700
Subject: [PATCH 1/3] Make `setup.py` consistent with conda recipe

Inconsistencies in `setup.py` can cause `pip check` to fail
on the conda package.
---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index bd283df..bb5feb9 100644
--- a/setup.py
+++ b/setup.py
@@ -28,7 +28,7 @@ def package_files(directory, prefixes, extensions):
     author_email="forsyth2@llnl.gov, golaz1@llnl.gov",
     description="Post-processing software for E3SM",
     python_requires=">=3.6",
-    intall_requires=["configobj", "jinja2"],
+    intall_requires=["configobj>=5.0.0,<6.0.0", "jinja2>=2.0.0"],
     packages=find_packages(include=["zppy", "zppy.*"]),
     package_data={"": data_files},
     entry_points={"console_scripts": ["zppy=zppy.__main__:main"]},

From b5ff07ae5d430a99676a9bc3c7cc7d164acf57d2 Mon Sep 17 00:00:00 2001
From: Tom Vo <tomvothecoder@gmail.com>
Date: Mon, 30 Jan 2023 12:22:44 -0800
Subject: [PATCH 2/3] Update GH Actions actions versions

---
 .github/workflows/build_workflow.yml   | 14 +++++++-------
 .github/workflows/release_workflow.yml | 10 +++++-----
 2 files changed, 12 insertions(+), 12 deletions(-)

diff --git a/.github/workflows/build_workflow.yml b/.github/workflows/build_workflow.yml
index 5814db3..76e6c48 100644
--- a/.github/workflows/build_workflow.yml
+++ b/.github/workflows/build_workflow.yml
@@ -28,10 +28,10 @@ jobs:
     timeout-minutes: 2
     steps:
       - name: Checkout Code Repository
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
 
       - name: Set up Python
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v4
         with:
           python-version: 3.9
 
@@ -39,7 +39,7 @@ jobs:
       # Getting only staged files can be tricky in case a new PR is opened
       # since the action is run on a branch in detached head state
       - name: Install and Run Pre-commit
-        uses: pre-commit/action@v2.0.0
+        uses: pre-commit/action@v3.0.0
 
   build:
     needs: check-jobs-to-skip
@@ -49,10 +49,10 @@ jobs:
       run:
         shell: bash -l {0}
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
 
       - name: Cache Conda
-        uses: actions/cache@v2
+        uses: actions/cache@v3
         env:
           CACHE_NUMBER: 0
         with:
@@ -99,12 +99,12 @@ jobs:
           fetch-depth: 0
 
       - name: Set up Python
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v4
         with:
           python-version: 3.9
 
       - name: Cache pip
-        uses: actions/cache@v2
+        uses: actions/cache@v3
         with:
           # This path is specific to Ubuntu
           path: ~/.cache/pip
diff --git a/.github/workflows/release_workflow.yml b/.github/workflows/release_workflow.yml
index 6c2f3c7..4c9c9ac 100644
--- a/.github/workflows/release_workflow.yml
+++ b/.github/workflows/release_workflow.yml
@@ -12,10 +12,10 @@ jobs:
       run:
         shell: bash -l {0}
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
 
       - name: Cache Conda
-        uses: actions/cache@v2
+        uses: actions/cache@v3
         env:
           CACHE_NUMBER: 0
         with:
@@ -62,18 +62,18 @@ jobs:
         shell: bash -l {0}
     timeout-minutes: 5
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         with:
           persist-credentials: false
           fetch-depth: 0
 
       - name: Set up Python
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v4
         with:
           python-version: 3.9
 
       - name: Cache pip
-        uses: actions/cache@v2
+        uses: actions/cache@v3
         with:
           # This path is specific to Ubuntu
           path: ~/.cache/pip

From d8be84f045926a1ba3ab04c5ff4593dd05be535c Mon Sep 17 00:00:00 2001
From: Tom Vo <tomvothecoder@gmail.com>
Date: Mon, 30 Jan 2023 12:25:46 -0800
Subject: [PATCH 3/3] Bump `isort` version to fix poetry error

---
 .pre-commit-config.yaml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/.pre-commit-config.yaml b/.pre-commit-config.yaml
index efc3ea9..177692e 100644
--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -18,8 +18,8 @@ repos:
       - id: black
 
   # Can run individually with `pre-commit run isort --all-files`
-  - repo: https://github.com/timothycrosley/isort
-    rev: 5.10.1
+  - repo: https://github.com/PyCQA/isort
+    rev: 5.12.0
     hooks:
       - id: isort
 
